### ISSUE TITLE
[GSoC] Project: Transfer Learning 

### ISSUE THREAD

**Issue by dzlab** [OpenMined/SyferText#16]
Transfer learning help in transferring knowledge from a source domain to a different target domain. In NLP, Transfer learning become ubiquitous as a way of reusing pre-trained language models and fine tuning them to a target task like Text Classification or Named Entity Recognition.
The goal of this project is to apply Transfer Learning to a private dataset using SyferText.

An example would be to train a BERT-based model for text auto-completion and personalized per user. A practical use case will be a question completion assistant that leverages the Quora questions dataset. We can fine tune an English Bert-based language model on Quora questions dataset (public) then personalize the model to end user inputs.

This project should not be too complicated, but requires some exploratory work to further define the task and choose dataset.

**Required Skills**:

Experience with NLP (e.g. Spacy, NTLK)
Experience with PyTorch
Familiarity (or willingness to become familiar) with SyferText
Familiarity (or willingness to become familiar) with PySyft

**Difficulty**: Medium

**Expectation**: A notebook and web (or mobile) demo. Bonus point for writing a blog post.

**Tasks**:
1. Familiarize yourself with SyferText: run the tutorials
2. Perform an initial exploratory analysis of the dataset (e.g. a Kaggle notebook)
3. Familiarize yourself with Transfer Learning:
3.1. Compare existing pre-trained Language Models: size, inference time
3.2. Choose a backbone model (e.g. BERT) and download the weights
3.3. Use the dataset to fine tune your model
3.4. Evaluate the performance of the fine-tuned model (e.g. using Blue score metric)
4. Try using the Syfer Text to do the same previous steps

The best way to share your progress on those tasks is through a GitHub repo.

Does this looks interesting to you?  then
* fill in this form https://mentorship.openmined.org
* join #gsoc channel on OpenMined and let us know.

**References**:
* Quora question dataset - [link](https://www.kaggle.com/c/quora-question-pairs)
* VisionAir: Using Federated Learning to estimate Air Quality using the Tensorflow API for Java - [link](https://blog.tensorflow.org/2020/02/visionair-using-federated-learning-to-estimate-airquality-tensorflow-api-java.html)
* Notebooks to get started with the Transformers library - [link](https://github.com/huggingface/transformers/tree/master/notebooks)

**Comment by carlodavid012** [comment]
I'd like to contribute to this

**Comment by AntonioLopardo** [comment]
I would also be very interested

**Comment by AshutoshFreak** [comment]
Hello,
I am Ashutosh Sahu from IIT Bhilai and am very much interested in Sequence Models and have some experience with projects in Transfer Learning like Neural Style Transfer, etc. Though I don't have any experience with PyTorch, but it can't be that hard (even if it is, I can do it!) to move from a Tensorflow background.
I would love to contribute to this project for betterment of both the community and gaining knowledge in different fields. I hope its not too late for joining this project!

However, I have some questions in mind-
1. Can someone please mention who is the mentor for this project? @dzlab are you? 
2. I will be starting the warm up tasks pretty soon. I would like to ask that where do I post my progress and proof of completion? Should I post it here or there is some separate place to monitor progress?

**Comment by dzlab** [comment]
@AshutoshFreak this is cool, happy to see you interested in this. Yes I'm the mentor, I've updated the project description, the best way is to have a GitHub repo for notebooks/code/markdown documents.

**Comment by arbazkhan971** [comment]
hey I am interested in this project and I know Transfer learning in NLP
and I have used BERT, RoBERTa, XLNet, XLM, DistilBERT IN DIFFERENT KAGGLE competitions 
and used them in my different projects
Let me know how to contribute I am excited to work in this problem

**Comment by arbazkhan971** [comment]
I am very comfortable with huggingface transformer

**Comment by dzlab** [comment]
@arbazkhan971  can you join the openmined slack channel of this project #gsoc_nlp_transfer_learning

**Comment by sukannyapurkayastha** [comment]
Hi, I am Sukannya Purkayastha, an MS (Research) student at IIT Kharagpur. I have prior experience in Machine learning and Deep Learning and have also co-authored papers at International machine learning venues. I have worked with various deep learning models such as Transformers, Seq2Seq Models nad also on fine-tuning language models such as BERT, etc. I would like to contribute to this project.

**Comment by dzlab** [comment]
@sukannyapurkayastha please join the slack channel #gsoc_nlp_transfer_learning

**Comment by Infernolia** [comment]
Hello! I am Aboli Marathe from India. I have joined the slack channel and would like to contribute to this project!

**Comment by dzlab** [comment]
@Infernolia did you introduce yourself in the channel? I will reply back and answer any question.

**Comment by Infernolia** [comment]
Yes, I shall. Thanks!

**Comment by pranshurastogi29** [comment]
hello @dzlab  i am interested in this project i will try to follow your views on how to persieve this task
my kaggle  - https://www.kaggle.com/pranshu29 just incase if you want to see my expertise
will keep on updating you on my progress

**Comment by dzlab** [comment]
@pranshurastogi29 sure, feel free to reach out when you have questions here or on slack

**Comment by codeboy5** [comment]
Hi @dzlab I am interested in working on this project ? how can I contribute ?

**Comment by dzlab** [comment]
@codeboy5 read the issue description to get started.

**Comment by AlanAboudib** [comment]
I am closing all GSOC issues. Any contribution should be push directly as PR
